nvtabular/column.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional, Text
+
+import tree
+
+from nvtabular.tag import DefaultTags
+
+
+@dataclass(frozen=True)
+class Column:
+    """"A Column with metadata. """
+
+    name: Text
+    tags: Optional[List[Text]] = field(default_factory=list)
+    properties: Optional[Dict[Text, Text]] = field(default_factory=dict)
+
+    def __post_init__(self):
+        if isinstance(self.tags, DefaultTags):
+            object.__setattr__(self, "tags", self.tags.value)
+
+    def __str__(self) -> str:
+        return self.name
+
+    def with_tags(self, tags, add=True) -> "Column":
+        if isinstance(tags, DefaultTags):
+            tags = tags.value
+        if not tags:
+            return self
+
+        tags = list(set(list(self.tags) + list(tags))) if add else tags
+
+        return Column(self.name, tags=tags, properties=self.properties)
+
+    def with_name(self, name) -> "Column":
+        return Column(name, tags=self.tags, properties=self.properties)
+
+    def with_properties(self, add=True, **properties) -> "Column":
```

Comment:
I feel like for production use-cases the properties could be super useful. We could think of some potential use-cases for it for the library itself as well.

Reply:
Yeah, I'm not opposed to adding properties in general, just wondering if there's anything currently relying on it. If not, it might be worth postponing to a future PR to keep the scope for this release as small as we can.

Reply:
Is one use case the dimensionality of embeddings? (like the tags would be 'categorical' but the properties would be 'max=embedding_dimensionality?' I could see this being useful for the hugectr flat embedding table if so

Reply:
I agree! We could also reason from the other way, since we want to allow to create a `ColumnGroup` from a schema we should make sure that we cover all fields there. As you can see here they allow to set the `annotation` field that next to tags also includes a comment and extra_metadata ([source](https://github.com/tensorflow/metadata/blob/master/tensorflow_metadata/proto/v0/schema.proto#L214-L224)).

Reply:
I'm cool with all of that, but will still probably opt to postpone adding properties until we're actually implementing a feature that requires them.